File: examples/srcs/shared/module_d/module_d.m
#import "module_d.h"
#import "srcs/shared/module_d/srcs_shared_module_d_module_d-Swift.h"

@implementation ModuleDObjc

- (NSString *)callSwiftMethod {
    return [ModuleDSwift new].calledByObjcMethod;
}

- (NSString *)calledBySwiftMethod {
    return @"i am objc, called by swift";
}

@end

File: examples/srcs/shared/module_d/module_d.swift
import Foundation

@objcMembers
public class ModuleDSwift : NSObject {

    public func callObjcMethod() -> String {
        let moduleDObjc = ModuleDObjc()
        return moduleDObjc.calledBySwiftMethod()
    }

    public func calledByObjcMethod() -> String {
        return "i am swift, called by objc"
    }

}